### DISCUSSION TITLE
Bad Power TuYa TS011F_plug_1

### DISCUSSION THREAD

**Discussion by casse-boubou** [Koenkk/zigbee2mqtt#13312] [Devices]
Hello,
I bought a TuYa plug and it is detected as TS011F_plug_1
All works well except for the Power function which I do not understand the calculation
If I'm not mistaken, the calculation for Power is: Power(W) = Voltage (V) x Current (A)
But the values reported for Power by the plug do not correspond to this calculation.
I tried with several different devices and each time the Power value is more than twice less than it should be. Only my fan comes close to the calculated real value.
Does anyone know why such a discrepancy between the calculated value and the displayed value? Am I missing something in the calculation? A characteristic of electricity that I do not take into account?
Thank you very much to all !
![Composition 1_00000](https://user-images.githubusercontent.com/21007415/181253134-56b0cb85-f1de-46b3-a5ab-8782b2b1a8d1.png)

**Comment by boernerm** [comment]
the same here

**Reply by kkossev** [reply]
Tuya TS011F plugs are sending only the active power, which depends on the power factor.  
Only in the cases, when you have a purely resistive load, the voltage multiplied by the amperage will equal the momentary power.
The formula for the Activer Power is actually  P = V x I x Cosθ , where  Cosθ is the [power factor](https://www.fluke.com/en/learn/blog/power-quality/power-factor-formula#:~:text=It%20is%20found%20by%20multiplying,power%20delivered%20to%20the%20circuit.) ( the voltage and amperage phase angle difference).
